pkg/cloud/instance.go

```diff
@@ -212,6 +212,14 @@ func (c *client) GetOrCreateVMInstance(
 
 	deployVMResp, err := c.cs.VirtualMachine.DeployVirtualMachine(p)
 	if err != nil {
+		// Just because an error was returned doesn't mean a (failed) VM wasn't created and will need to be dealt with.
+		// Regretfully the deployVMResp may be nil, so we need to get the VM ID with a separate query, so we
+		// can return it to the caller, so they can clean it up.
+		listVirtualMachineParams := c.cs.VirtualMachine.NewListVirtualMachinesParams()
+		listVirtualMachineParams.SetName(csMachine.Name)
```

Comment:
Is the name unique enough that we can be sure we're only going to get the one machine back here?

Reply:
I did a preliminary test of attempting to create a VM with the same name as on that existed, and it was rejected.  But in retrospect, it was in the same zone/domain as the existing one, and I can't say that there are circumstance where name along is not sufficient. I'll narrow that query to use as much as we know about the VM we attempted to create.